proto/poktroll/shared/supplier.proto

```diff
@@ -10,15 +10,24 @@ import "poktroll/shared/service.proto";
 
 // Supplier is the type defining the actor in Pocket Network that provides RPC services.
 message Supplier {
-  string address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // The Bech32 address of the supplier using cosmos' ScalarDescriptor to ensure deterministic encoding
-  cosmos.base.v1beta1.Coin stake = 2; // The total amount of uPOKT the supplier has staked
-  repeated SupplierServiceConfig services = 3; // The service configs this supplier can support
+  // The address of the owner (i.e. staker, custodial) that owns the funds for staking.
+  // By default, this address is the one that receives all the rewards unless owtherwise specified.
+  // The owner can initially stake the supplier and unstake it.
+  // All other configurations are managed by the operator.
+  string owner_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"]; // Bech32 cosmos address
+  // The operator address of the supplier that operates it.
+  // The operator address can update the supplier's configurations excluding the owner
+  // and operator addresses which do not change over the supplier's lifespan.
```

Comment:
> and operator addresses which do not change over the supplier's lifespan.

This is incorrect.

The operator can:
- Update operator address

The owner can:
- Update operator address
- Update owner address.

Reply:
I have added this restriction to simplify the `Supplier` management.

Given that:
1. Changing the operator address mid-session would have the same effect as changing other service configs, which raises the need to delay the update til the next session.

2. The `OperatorAddress` is the `Supplier` identifier, if the owner is allowed to change it, it would need to specify the old `OperatorAddress` so we would know which supplier to update when calling `GetSupplier` (we actually have to delete then set the `Supplier` entry in the `Supplier` KVStore). Adding an `old_operator_address` to the staking config file is far from ideal ux wise.

Having the owner being able to update the `OwnerAddress` is fine though.

Reply:
> Having the owner being able to update the OwnerAddress is fine though.

Let's make this clearer.

Reply:
> Changing the operator address mid-session would have the same effect as changing other service configs, which raises the need to delay the update til the next session.

Add one line making it clearer why operator address is immutable.

Reply:
Per this comment: https://github.com/pokt-network/poktroll/pull/716#discussion_r1707865400

Can you please evaluate if there is any difference with how it works in Morse?

Reply:
Looking directly into the code, these are the current differences.

|  | Morse | Shannon |
|---|---|---|
| Id | Operator address | Operator address |
| Initial stake | Owner and Operator | Owner only |
| Update | Owner and operator | Operator only |
| Update owner| Owner | None, must unstake->stake |
| Update operator| Owner and Operator (see Note) | None, must unstake->stake |
| Unstake | Owner and Operator | Owner only (can easily upgrade to both) |

_Note that from reading the code, ther seems to be no code path that can update the operator address, since the message would contain a new validator address which would not be found when doing `k.GetValidator(ctx, validator.Address)`. I may be missing something though_